MahApps/MahApps.Metro
# Add TreeListView

As already mentioned in #2604, I was writing a TreeListView for a personal project which is basically a TreeView with columns (like the ListView).
![TreeListView](http://i.imgur.com/b7QqbAo.png)

One of the biggest problems with this control is that you have to somehow set the tree space to the first column and not to the whole GridViewRowPresenter. I solved that by modifying the CellTemplate: I wrapped the original template in a ContentPresenter, set the Margin of the ContentPresenter and finally set the ContentPresenter as the new CellTemplate.

It is always the first column which has the expanding button. If the first column is changed, the template will be reset for and the new first column will be modified.

If you want to test that control:

``` xaml
        <controls:TreeListView ChildrenPropertyName="Children" ItemsSource="{Binding Collection}">
            <controls:TreeListView.View>
                <GridView>
                    
                </GridView>
            </controls:TreeListView.View>
        </controls:TreeListView>
```

I should mention the property `ChildrenPropertyName`. I made the `ItemTemplate` property read only so the developer doesn't get confused why that doesn't work. The TreeListView just doesn't support this property, but because WPF handles `HierarchicalDataTemplate`s really well, the `ChildrenPropertyName` just sets the `HierarchicalDataTemplate` to a new template with the `ItemsSource` = a binding with the `ChildrenPropertyName` as path.
There are two modifications possible:
1. Change the `ChildrenPropertyName` to type Binding (instead of string)
2. Remove the `ChildrenPropertyName` property and the user has to set an empty ItemTemplate by himself.
I don't know which would be better, I like it with a simple property.
